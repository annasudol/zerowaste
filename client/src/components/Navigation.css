
    /* drawer menu pane - note the 0px width */ 
    
    .closed-nav { 
        position: fixed; 
        top: 0; 
        right:-200px; 
        height: 100%; 
        width: 200px; 

    .opened-nav {
        position: fixed; 
        top: 0; 
        right:0; 
        height: 100%; 
        width: 200px;
        background: #D8C3A5; 
    }

    .nav {
        width: 100px;
        position: relative;
        left: 30px;
        top: 50px;
    }

    .nav ul {
        list-style-type:none;
    }

    
    /* actual page content pane */ 
    

    
    /* checked styles (menu open state) */ 
    
    #drawer-toggle:checked ~ #drawer-toggle-label { 
    height: 100%; 
    width: calc(100% - 300px); 
    background: rgba(255,255,255,.8); 
    } 
    
    #drawer-toggle:checked ~ #drawer-toggle-label, 
    #drawer-toggle:checked ~ header { 
    left: 300px; 
    } 
    
    #drawer-toggle:checked ~ #drawer { 
    left: 0px; 
    } 
    
    #drawer-toggle:checked ~ #page-content { 
    margin-left: 300px; 
    }
    
    /* Menu item styles */ 
    
    #drawer ul { 
    list-style-type:none; 
    } 
    
    #drawer ul a { 
    display:block; 
    padding:10px; 
    color:#c7c7c7; 
    text-decoration:none; 
    } 
    
    #drawer ul a:hover { 
    color:white; 
    } 
    
    /* Responsive MQ */ 
/*     
    @media all and (max-width:350px) { 
    
        #drawer-toggle:checked ~ #drawer-toggle-label { 
        height: 100%; 
        width: 50px; 
        } 
        
        #drawer-toggle:checked ~ #drawer-toggle-label, 
        #drawer-toggle:checked ~ header { 
        left: calc(100% - 50px); 
        } 
        
        #drawer-toggle:checked ~ #drawer { 
        width: calc(100% - 50px); 
        padding: 20px; 
        } 
        
        #drawer-toggle:checked ~ #page-content { 
        margin-left: calc(100% - 50px); 
        } 
        
    } */