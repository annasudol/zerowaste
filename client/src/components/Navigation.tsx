import * as React from 'react';
import cx from 'classnames';
import { AppRoutes } from '../../routes';
import { useHistory } from 'react-router';
import { Button } from 'antd';
import { Link } from 'react-router-dom';
import { MenuUnfoldOutlined } from '@ant-design/icons';
import CancelIcon from '@material-ui/icons/Cancel';
import { useAuth } from "../components";

export const Navigation: React.FC = (): React.ReactElement => {
    const [open, toggleOpen] = React.useState<boolean>(true);
    const { userLogged, logoutUser } = useAuth();
    const history = useHistory();

    const logout = (): void => {
        logoutUser()
        history.push(`${AppRoutes.Home}`)
    }


    const loggedIn = (<>
        <ul>
            <li className='mb-4'><Link to={AppRoutes.Home} className='pt-2 pb-2'>Home</Link></li>
            <li className='mb-4'><Link to={AppRoutes.AddRecipe} className='pt-2 pb-2'>Add Recipe</Link></li>
            <li className='mb-4'><Link to={AppRoutes.User} className='pt-2 pb-2'>Account</Link></li>
        </ul>
        <Button onClick={logout} type='primary' danger>logout</Button>
    </>);
    const loggedOut = (<ul>
        <li className='mb-4'><Link to={AppRoutes.Home} className='pt-2 pb-2'>Home</Link></li>
        <li className='mb-4'><Link to={AppRoutes.Login} className='pt-2 pb-2'>Login</Link></li>
        <li className='mb-4'><Link to={AppRoutes.SignUp} className='pt-2 pb-2'>Sign up</Link></li>
    </ul>)

    return (
        <>
            {!open && <Button onClick={(): any => toggleOpen(true)} type='primary' icon={<MenuUnfoldOutlined />} size='large' shape='circle' className='nav-btn mb-10 lg:hidden' />}
            <div className={cx(open ? 'opened-nav' : 'closed-nav')}>
                <nav className='nav'>
                    <Button onClick={(): any => toggleOpen(false)} type='link' className='outline-none border-none bg-transparent mb-4 p-0 border-0 lg:hidden'><CancelIcon style={{ color: '#fff' }} /></Button>
                    {userLogged ? loggedIn : loggedOut}
                </nav>
            </div>
        </>
    );
}