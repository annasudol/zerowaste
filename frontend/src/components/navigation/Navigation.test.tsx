import { shallow } from 'enzyme';
import React from 'react';
import { Navigation } from './Navigation';
import toJson from 'enzyme-to-json';
import * as router from 'react-router';

// import { renderHook } from '@testing-library/react-hooks';

// require('../../hooks');
// const useAuth = require('../../hooks').useAuth as jest.Mock;

describe('component', (): void => {
  describe('LoginSignUpForm', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(<Navigation />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('responds for closing and opening navigation', (): void => {
      const wrapper = shallow(<Navigation />);
      expect(wrapper.children().at(1).hasClass('opened-nav')).toBe(true);
      expect(wrapper.children().at(1).hasClass('closed-nav')).toBe(false);

      // closing navigation
      wrapper.find('Button[type="link"]').simulate('click');
      expect(wrapper.children().at(1).hasClass('opened-nav')).toBe(false);
      expect(wrapper.children().at(1).hasClass('closed-nav')).toBe(true);

      // opening navigation
      wrapper.find('Button[shape="circle"]').simulate('click');
      expect(wrapper.children().at(1).hasClass('opened-nav')).toBe(true);
      expect(wrapper.children().at(1).hasClass('closed-nav')).toBe(false);

    });
  });


});