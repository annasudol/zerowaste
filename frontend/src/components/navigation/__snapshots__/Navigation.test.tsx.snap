// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component LoginSignUpForm match snapshot 1`] = `
<Fragment>
  <div
    className="nav-btn"
  />
  <div
    className="opened-nav"
  >
    <nav
      className="nav"
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="link"
      >
        <CancelIcon
          style={
            Object {
              "color": "#fff",
            }
          }
        />
      </Button>
      <ul>
        <li
          className="mb-4"
        >
          <Link
            className="pt-2 pb-2"
            to="/"
          >
            Home
          </Link>
        </li>
        <li
          className="mb-4"
        >
          <Link
            className="pt-2 pb-2"
            to="/login"
          >
            Login
          </Link>
        </li>
        <li
          className="mb-4"
        >
          <Link
            className="pt-2 pb-2"
            to="/signUp"
          >
            Sign up
          </Link>
        </li>
      </ul>
    </nav>
  </div>
</Fragment>
`;
